--- main/src/FileProvider.cpp
+++ main/src/FileProvider.cpp
@@ -315,6 +315,11 @@
       file->isError = true;
       BELL_LOG(error, LOG_TAG, "Could not request audio key, err={}",
                requestRes.error());
+      
+      if (requestRes.error() == std::errc::operation_not_permitted) {
+          BELL_LOG(error, LOG_TAG, "AP connection is dead, sleeping to prevent spam loop...");
+          std::this_thread::sleep_for(std::chrono::seconds(5));
+      }
       eventLoop->post(EventLoop::EventType::FILE_PROVIDED, std::move(*file));
       return;
     }
